### PR TITLE
Hw 3 2

### PR DESCRIPTION
**Pull request by IgorBochkor**
added solution

### CODE REVIEW

File: src/main/java/core/basesyntax/Figure.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+import core.basesyntax.constant.Color;
+import core.basesyntax.producers.ColorProducer;
+
+public abstract class Figure implements Drawable {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Figure.java:6]: you figure should have field - color (it's a state for each object )

File: src/main/java/core/basesyntax/Figure.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    public abstract double getArea();
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Figure.java:7]: move method to an interface - that's behavior, interfaces are used to represent behavior

File: src/main/java/core/basesyntax/producers/FigureProducer.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import core.basesyntax.Figure;
+import core.basesyntax.shape.Circle;
+import core.basesyntax.shape.IsoscelesTrapezoid;
+import core.basesyntax.shape.Rectangle;
+import core.basesyntax.shape.RightTriangle;
+import core.basesyntax.shape.Square;
+import java.util.Random;
+
+public class FigureProducer {
+    public static final int COUNT_OF_FIGURE_IMPLEMENTATIONS = 5;
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/producers/FigureProducer.java:12]:
```suggestion
    public static final int COUNT_OF_FIGURE = 5;
```
Figure is a class, we don't implement classes only interfaces

File: src/main/java/core/basesyntax/producers/NumberProducer.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package core.basesyntax.producers;
+
+import java.util.Random;
+
+public class NumberProducer {
+    public static int getNumber() {
+        return new Random().nextInt(15) + 1;
+    }
+
+    public static int getNumberListFigure() {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/producers/NumberProducer.java:10]:
```suggestion
    public static int getFigureListIndex() {
```

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Figure.java (hunk 1 above)
Comment (IgorBochkor) [reply src/main/java/core/basesyntax/Figure.java:6]: done

File: src/main/java/core/basesyntax/Figure.java (hunk 2 above)
Comment (IgorBochkor) [reply src/main/java/core/basesyntax/Figure.java]: done!

File: src/main/java/core/basesyntax/producers/FigureProducer.java (hunk 3 above)
Comment (IgorBochkor) [reply src/main/java/core/basesyntax/producers/FigureProducer.java]: done

File: src/main/java/core/basesyntax/producers/NumberProducer.java (hunk 4 above)
Comment (IgorBochkor) [reply src/main/java/core/basesyntax/producers/NumberProducer.java]: done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request closed without merging**